src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -472,38 +502,39 @@ export function validateComponentFormData(
   return null;
 }
 
-export function getSchemaPart(dataModelPath: string[], subSchema: any, mainSchema: any) {
-  const dataModelRoot = dataModelPath[0];
-  if (subSchema.properties && subSchema.properties[dataModelRoot] && dataModelPath && dataModelPath.length !== 0) {
-    const localRootElement = subSchema.properties[dataModelRoot];
-    if (localRootElement.$ref) {
-      const childSchemaPtr = JsonPointer.compile(localRootElement.$ref.substr(1));
-      return getSchemaPart(dataModelPath.slice(1), childSchemaPtr.get(mainSchema), mainSchema);
-    }
-    if (localRootElement.items && localRootElement.items.$ref) {
-      const childSchemaPtr = JsonPointer.compile(localRootElement.items.$ref.substr(1));
-      return getSchemaPart(dataModelPath.slice(1), childSchemaPtr.get(mainSchema), mainSchema);
-    }
-    return localRootElement;
-  }
+/**
+ * Wrapper method around getSchemaPart for schemas made with our old generator tool
+ * @param schemaPath the path, format #/properties/model/properties/person/properties/name/maxLength
+ * @param mainSchema the main schema to get part from
+ * @param rootElementPath the subschema to get part from
+ * @returns the part, or null if not found
+ */
+export function getSchemaPartOldGenerator(schemaPath: string, mainSchema: object, rootElementPath: string): any {
+  // for old generators we can have a ref to a definition that is placed outside of the subSchema we validate against.
+  // if we are looking for #/definitons/x we search in main schema
 
-  if (subSchema.allOf) {
-    let tmpSchema: any = {};
-    subSchema.allOf.forEach((element) => {
-      tmpSchema = {
-        ...tmpSchema,
-        ...element,
-      };
-    });
-    return getSchemaPart(dataModelPath, tmpSchema, mainSchema);
+  if (schemaPath.startsWith('#/definitions/')) {
+    return getSchemaPart(schemaPath, mainSchema);
   }
+  // all other in sub schema
+  return getSchemaPart(schemaPath, getSchemaPart(`${rootElementPath}/#`, mainSchema));
+}
 
-  if (subSchema.$ref) {
-    const ptr = JsonPointer.compile(subSchema.$ref.substr(1));
-    return getSchemaPart(dataModelPath.slice(1), ptr.get(mainSchema), mainSchema);
+/**
+ * Gets a json schema part by a schema patch
+ * @param schemaPath the path, format #/properties/model/properties/person/properties/name/maxLength
+ * @param jsonSchema the json schema to get part from
+ * @returns the part, or null if not found
+ */
+export function getSchemaPart(schemaPath: string, jsonSchema: object): any {
+  try {
+    // want to transform path example format to to /properties/model/properties/person/properties/name
+    const pointer = schemaPath.substr(1).split('/').slice(0, -1).join('/');
```

Comment:
No big deal, but if you declare the pointer before the try block, you can inspect it by setting a break point in the catch block.
This is only useful for debugging and shouldn't matter that much.

Reply:
Unsure if I'm missing something, but setting a breakpoint inside the try block is totally possible as it is?
![image](https://user-images.githubusercontent.com/10416981/134013949-b754dd6e-0630-41cc-b3cb-2ca4b83e318c.png)

Reply:
Yes.
What I meant was that if you want to only break in the cases where `try` block fails (placing the break point inside the `catch` block,) then the `pointer` variable is no longer in a accessible scope.